Description: Do not keep swtpm pidfile around after stopping qemu vm
Author: Martin Kletzander <mkletzan@redhat.com>
Origin: upstream, https://gitlab.com/libvirt/libvirt/-/commit/3c2d06d78e1bd2d9298276b44a6ab09cc3b36e5a
Bug: https://bugzilla.redhat.com/show_bug.cgi?id=2111301
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/1997269
Last-Update: 2022-11-21
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/qemu/qemu_tpm.c
+++ b/src/qemu/qemu_tpm.c
@@ -793,28 +793,25 @@
     g_autofree char *pathname = NULL;
     g_autofree char *errbuf = NULL;
     g_autofree char *swtpm_ioctl = virTPMGetSwtpmIoctl();
+    g_autofree char *pidfile = qemuTPMEmulatorPidFileBuildPath(swtpmStateDir,
+        shortName);
 
-    if (!swtpm_ioctl)
-        return;
+    if (swtpm_ioctl &&
+        (pathname = qemuTPMEmulatorSocketBuildPath(swtpmStateDir, shortName)) &&
+        virFileExists(pathname)) {
 
-    if (!(pathname = qemuTPMEmulatorSocketBuildPath(swtpmStateDir, shortName)))
-        return;
+        cmd = virCommandNewArgList(swtpm_ioctl, "--unix", pathname, "-s", NULL);
 
-    if (!virFileExists(pathname))
-        return;
+        virCommandSetErrorBuffer(cmd, &errbuf);
 
-    cmd = virCommandNew(swtpm_ioctl);
-    if (!cmd)
-        return;
+        ignore_value(virCommandRun(cmd, NULL));
 
-    virCommandAddArgList(cmd, "--unix", pathname, "-s", NULL);
+        /* clean up the socket */
+        unlink(pathname);
+    }
 
-    virCommandSetErrorBuffer(cmd, &errbuf);
-
-    ignore_value(virCommandRun(cmd, NULL));
-
-    /* clean up the socket */
-    unlink(pathname);
+    if (pidfile)
+        virPidFileForceCleanupPath(pidfile);
 }
 
 
